### PR TITLE
[HF] Implement save_to_hf_hub for Model and Tokenizer

### PR DESCRIPTION
**Pull request by sijunhe**
### PR types
New features

### PR changes
APIs

### Description
- implement save_to_hf_hub for Model and Tokenizer
```
from paddlenlp.transformers import AutoModel, AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("PaddlePaddle/ci-test-ernie-model", from_hf_hub=True)
model = AutoModel.from_pretrained("PaddlePaddle/ci-test-ernie-model", from_hf_hub=True)

tokenizer.save_to_hf_hub("ci-test-ernie-model")
model.save_to_hf_hub("ci-test-ernie-model")
```

### CODE REVIEW

File: paddlenlp/transformers/model_utils.py
Comment (sijunhe) [inline paddlenlp/transformers/model_utils.py:708]: change is here

File: paddlenlp/transformers/tokenizer_utils_base.py
Comment (sijunhe) [inline paddlenlp/transformers/tokenizer_utils_base.py:1750]: change is here

File: paddlenlp/transformers/model_utils.py
Comment (sijunhe) [inline paddlenlp/transformers/model_utils.py]: @wj-Mcat 
this line was marked as unused variable definition. Please check whether it's safe to remove

Comment (wj-Mcat) [reply paddlenlp/transformers/model_utils.py]: this variable is not used, so it shoule be removed.

Comment (wj-Mcat) [reply paddlenlp/transformers/model_utils.py]: I have removed it in another pr.  https://github.com/PaddlePaddle/PaddleNLP/pull/3965

File: paddlenlp/transformers/tokenizer_utils_base.py
Comment (wj-Mcat) [inline paddlenlp/transformers/tokenizer_utils_base.py]: 我这边有几个问题：
1. 如果 repo 里面本身就有大文件，clone下载就会很慢？
2. 如果在 clone 的过程由于时间过程失败了，那由于是创建的临时文件夹每次都得重来，是否会浪费时间，如果这个 tmp_dir能够指定是否更好呢？

File: paddlenlp/transformers/model_utils.py
Comment (wj-Mcat) [inline paddlenlp/transformers/model_utils.py]: `git lfs`是需要安装对应的包的，这里是否需要提供检测lsf是否存在并抛一个安装的 warning 呢？

File: paddlenlp/transformers/model_utils.py
Comment (sijunhe) [reply paddlenlp/transformers/model_utils.py]: ok. i'll wait until you merge your PR in.

File: paddlenlp/transformers/tokenizer_utils_base.py
Comment (sijunhe) [reply paddlenlp/transformers/tokenizer_utils_base.py]: 你说的有道理。我新的实现取消了clone, 并且更换为http request

File: paddlenlp/transformers/model_utils.py
Comment (sijunhe) [reply paddlenlp/transformers/model_utils.py]: 已替换为http request

**Review by osanseviero — DISMISSED** [review]
Looks very nice!

File: paddlenlp/transformers/model_utils.py
Comment (Wauplin) [inline paddlenlp/transformers/model_utils.py]: There is no need to check for the token manually anymore.
If the user is not logged in and try to use `create_repo` it would raise a `ValueError`.
```
ValueError: Token is required (write-access action) but no token found. You need to provide a token or be logged in to Hugging Face with `huggingface-cli login` or `huggingface_hub.login`. See https://huggingface.co/settings/tokens.
```

The only need for an explicit `token=token` is if you set `token` as an argument of `save_to_hf_hub`, which is not the case here. So you can simplify the logic to:

```suggestion
        repo_url = create_repo(repo_id, private=private, exist_ok=True)
```

File: paddlenlp/transformers/tokenizer_utils_base.py
Comment (Wauplin) [inline paddlenlp/transformers/tokenizer_utils_base.py]: Same as above. Checking for `token` is not needed.

```suggestion
        repo_url = create_repo(repo_id, private=private, exist_ok=True)
```

**Review by Wauplin — COMMENTED** [review]
Hi @sijunhe , PR looks good to me ! :) 
I have added 2 comments as you don't need to check for `token` in the `HfFolder` (that's a bit too "low-level" for an external library).

File: paddlenlp/transformers/model_utils.py
Comment (sijunhe) [reply paddlenlp/transformers/model_utils.py]: Thanks for the suggestion

File: paddlenlp/transformers/tokenizer_utils_base.py
Comment (Wauplin) [inline paddlenlp/transformers/tokenizer_utils_base.py]:
```suggestion
```

File: paddlenlp/transformers/model_utils.py
Comment (Wauplin) [inline paddlenlp/transformers/model_utils.py]:
```suggestion
```

**Review by Wauplin — COMMENTED** [review]
(nit) `HfFolder` is not used anymore. Sorry, forgot about it in previous review :)

**Review by wj-Mcat — APPROVED** [review]
LGTM

**Comment by sijunhe** [comment]
Awesome! Thanks for so a quick turnaround!

**Pull request merged**
